demo/electron/src/components/VoiceTimer.vue

```diff
@@ -1,143 +1,221 @@
-      isError: false,
-      isLoaded: false,
-      isListening: false,
-      isTalking: false,
-      context64: CLOCK_EN_64,
-      info: null,
-      engine: null,
-      factory: PicovoiceWorkerFactoryEn,
-      factoryArgs: {
-        accessKey: "",
-        start: true,
-        porcupineKeyword: 'Picovoice',
-        rhinoContext: {
-          base64: CLOCK_EN_64,
-        },
-      },
+  setup() {
+    const { state, init, start, release } = usePicovoice();
+
+    const accessKey = ref("");
+    const engine = ref<"ppn" | "rhn">("ppn");
+    const interval = ref<any>(null);
+    const timeInitial = ref(0);
+    const timeRemaining = ref(0);
+    const timeRemainingDisplay = ref("");
+
+    const updateAccessKey = (event: any) => {
+      accessKey.value = event.target.value;
     };
-  },
-  methods: {
-    initEngine: function (event) {
-      this.factoryArgs.accessKey = event.target.value;
-      this.isError = false;
-      this.isLoaded = false;
-      this.isListening = false;
-      this.$refs.picovoice.initEngine();
-    },
-    start: function () {
-      if (this.$refs.picovoice.start()) {
-        this.isListening = !this.isListening;
+
+    const initPicovoice = async () => {
+      await init(
+        accessKey.value,
+        { builtin: BuiltInKeyword.Computer },
+        { publicPath: "porcupine_params.pv", forceWrite: true },
+        { publicPath: "clock_wasm.rhn", forceWrite: true },
+        { publicPath: "rhino_params.pv", forceWrite: true }
+      );
+      await start();
+    };
+
+    onBeforeUnmount(() => {
+      release();
+    });
+
+    const startTimer = () => {
+      if (interval.value !== null) {
+        clearInterval(interval.value);
+      }
+      interval.value = setInterval(() => {
+        timeRemaining.value = timeRemaining.value - 1;
+        if (timeRemaining.value <= 0) {
+          clearInterval(interval.value);
+        }
+        const hours = Math.floor(timeRemaining.value / 3600);
+        const minutes = Math.floor((timeRemaining.value - hours * 3600) / 60);
+        const seconds = timeRemaining.value - hours * 3600 - minutes * 60;
+        timeRemainingDisplay.value = `${hours
+          .toString()
+          .padStart(2, "0")}:${minutes.toString().padStart(2, "0")}:${seconds
+          .toString()
+          .padStart(2, "0")}`;
+      }, 1000);
+    };
+
+    const pauseTimer = () => {
+      if (interval.value !== null) {
+        clearInterval(interval.value);
+        interval.value = null;
+      }
+    };
+
+    const stopTimer = () => {
+      if (interval.value !== null) {
+        clearInterval(interval.value);
+        interval.value = null;
+        timeRemaining.value = 0;
       }
-    },
-    pause: function () {
-      if (this.$refs.picovoice.pause()) {
-        this.isListening = !this.isListening;
+    };
+
+    watch(
+      () => state.wakeWordDetection,
+      (wakeWordDetection) => {
+        if (wakeWordDetection !== null) {
+          engine.value = "rhn";
```

Comment:
I don't know much about ow the Picovoice Vue SDK works but I feel like this should be handled within the SDK no?

Reply:
I don't think we handle it inside sdks

Reply:
nvm I think I know wym, can handle it without setting the value